sdk/src/types/block/semantic.rs

```diff
@@ -342,6 +313,26 @@ impl<'a> SemanticValidationContext<'a> {
                     .checked_add(consumed_native_token.amount())
                     .ok_or(Error::ConsumedNativeTokensAmountOverflow)?;
             }
+
+            if let Some(unlocks) = self.unlocks {
+                if unlocks.len() != self.inputs.len() {
+                    return Ok(Some(TransactionFailureReason::InvalidInputUnlock));
+                }
+
+                let unlock = &unlocks[index];
```

Comment:
Should we do a safe get here?

Reply:
There is a check that unlocks and inputs have the same length right before. And the index comes from iterating the inputs.  Do you still prefer get + unwrap?

Reply:
Ah, that's fine then